astropy/config/configuration.py

```diff
@@ -852,3 +802,59 @@ def update_default_config(pkg, default_cfg_dir_or_fn, version=None, rootname='as
             return True
 
     return False
+
+
+def create_config_file(pkg, rootname='astropy', overwrite=False):
+    """
+    Create the default configuration file for the specified package.
+    If the file already exists, it is updated only if it has not been
+    modified.  Otherwise the ``overwrite`` flag is needed to overwrite it.
+
+    Parameters
+    ----------
+    pkg : str
+        The package to be updated.
+    rootname : str
+        Name of the root configuration directory.
+    overwrite : bool
+        Force updating the file if it already exists.
+
+    Returns
+    -------
+    updated : bool
+        If the profile was updated, `True`, otherwise `False`.
+
+    """
+
+    # local import to prevent using the logger before it is configured
+    from astropy.logger import log
+
+    cfgfn = get_config_filename(pkg, rootname=rootname)
+
+    # generate the default config template
+    template_content = io.StringIO()
+    generate_config(pkg, template_content)
+    template_content.seek(0)
+    template_content = template_content.read()
+
+    doupdate = True
+
+    # if the file already exists, check that it has not been modified
+    if cfgfn is not None and path.exists(cfgfn):
+        with open(cfgfn, 'rt', encoding='latin-1') as fd:
```

Comment:
Why does it use latin-1?

Reply:
No idea. I guess this comes from the old times with Python 2 etc. Since the plan is to remove this function in the future, I guess we can keep it as it is for now.